runtime/interpreter/value.go

```diff
@@ -7147,3 +7149,130 @@ func (v LinkValue) String() string {
 		v.TargetPath.String(),
 	)
 }
+
+// BuiltinCompositeValue
+type BuiltinCompositeValue struct {
+	Fields     *StringValueOrderedMap
+	staticType StaticType
+	structType *sema.BuiltinCompositeType
+}
+
+func NewBuiltinCompositeValue(structType *sema.BuiltinCompositeType, fields *StringValueOrderedMap) *BuiltinCompositeValue {
+	if fields == nil {
+		fields = NewStringValueOrderedMap()
+	}
+
+	return &BuiltinCompositeValue{
+		Fields:     fields,
+		structType: structType,
+		staticType: ConvertSemaToPrimitiveStaticType(structType),
+	}
+}
+
+func (*BuiltinCompositeValue) IsValue() {}
+
+func (v *BuiltinCompositeValue) Accept(interpreter *Interpreter, visitor Visitor) {
+	visitor.VisitBuiltinCompositeValue(interpreter, v)
+}
+
+func (v *BuiltinCompositeValue) DynamicType(_ *Interpreter) DynamicType {
+	return BuiltinCompositeDynamicType{v.structType}
+}
+
+func (v *BuiltinCompositeValue) StaticType() StaticType {
+	return v.staticType
+}
+
+func (v *BuiltinCompositeValue) Copy() Value {
+	return v
```

Comment:
This should return a copy, not a pointer to the same value. For example, all the fields need to be copied. See `CompositeValue.Copy`.